### PR TITLE
Update docs to describe how s3:// URLS does in fact support endpoint, region, and profile for upload

### PR DESCRIPTION
**Pull request by grahamc**
Improvements to https://github.com/NixOS/nix/pull/2319 after realizing I was wrong. cc @copumpkin 

I've tested all these with AWS S3 (copy-pasted these exact policies) and also DigitalOcean Spaces.

Rendered: https://screenshotscdn.firefoxusercontent.com/images/aa56ce00-b8a9-4616-b701-8ff50c0fc651.png

### CODE REVIEW

File: doc/manual/packages/s3-substituter.xml [hunk 1]
```diff
@@ -91,69 +123,49 @@ fetch prebuilt binaries from <uri>cache.nixos.org</uri>.</para>
 
   <para>Nix support fully supports writing to Amazon S3 and S3
   compatible buckets. The binary cache URL for our example bucket will
-  be <uri>s3://example-bucket</uri>.</para>
+  be <uri>s3://example-nix-bucket</uri>.</para>
```
Comment (manveru) [inline doc/manual/packages/s3-substituter.xml]: that was `example-nix-cache` above.

Comment (grahamc) [reply doc/manual/packages/s3-substituter.xml]: Fixed!

**Comment by grahamc** [comment]
Also, this one should definitely be backported :)

File: doc/manual/packages/s3-substituter.xml [hunk 2]
```diff
@@ -24,65 +65,56 @@ fetch prebuilt binaries from <uri>cache.nixos.org</uri>.</para>
...
-    }
-  ]
+    "Id": "DirectReads",
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "AlowDirectReads",
+            "Action": [
+                "s3:GetObject",
+                "s3:GetBucketLocation"
```
Comment (edolstra) [inline doc/manual/packages/s3-substituter.xml]: `GetBucketLocation` is no longer needed.

File: doc/manual/packages/s3-substituter.xml [hunk 3]
```diff
@@ -91,69 +123,49 @@ fetch prebuilt binaries from <uri>cache.nixos.org</uri>.</para>
...
-      path-based addressing instead of virtual host based
-      addressing.</para></note>
-    </listitem>
-    </varlistentry>
-  </variablelist>
-
-  <example><title>Uploading with non-default credential profile for Amazon S3</title>
-    <para><command>nix copy --to ssh://machine nixpkgs.hello s3://example-bucket?profile=cache-upload</command></para>
+  <para>Your account will need the following IAM policy to
+  upload to the cache:</para>
```
Comment (edolstra) [inline doc/manual/packages/s3-substituter.xml]: It's a bit strong to say that it *needs* this IAM policy. Rather the user needs certain rights which may be accomplished via this policy.

Comment (grahamc) [reply doc/manual/packages/s3-substituter.xml]: Fixed!

File: doc/manual/packages/s3-substituter.xml (hunk 2 above)
Comment (grahamc) [reply doc/manual/packages/s3-substituter.xml]: Fixed!

**Pull request merged**